File: Retrofit2WE/app/src/main/java/com/example/retrofit2_we/DataViewHolder.java
package com.example.retrofit2_we;

import android.view.View;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

public class DataViewHolder extends RecyclerView.ViewHolder {
    private TextView tvName, tvEmail, tvBody;

    public DataViewHolder(@NonNull View itemView) {
        super(itemView);
        initViews(itemView);

    }

    private void initViews(View itemView) {
        tvName = itemView.findViewById(R.id.tvName);
        tvEmail = itemView.findViewById(R.id.tvEmail);
        tvBody = itemView.findViewById(R.id.tvBody);

    }
    public void setData(ResponseModel responseModel){
        tvName.setText("Name: "+responseModel.getName());
        tvEmail.setText("Email: "+responseModel.getEmail());
        tvBody.setText("Body: "+responseModel.getBody());

    }
}

File: Kotlin Projects/KotlinScopingFunctionsYOU/app/src/main/java/com/example/kotlinscopingfunctionsyou/DataViewHolder.kt
package com.example.kotlinscopingfunctionsyou

import android.app.AlertDialog
import android.view.View
import androidx.recyclerview.widget.RecyclerView
import com.bumptech.glide.Glide
import kotlinx.android.synthetic.main.item_layout.view.*

class DataViewHolder(private val view: View) : RecyclerView.ViewHolder(view) {

    fun setData(dataModel: DataModel) {
        view.apply {
            tvEmail.text = dataModel.email
            tvID.text = dataModel.id.toString()
            tvName.text = dataModel.firstName
            Glide.with(ivProfile).load(dataModel.avatar).into(ivProfile)

            cardView.setOnClickListener {
                val  alertDialog = AlertDialog.Builder(context)
                alertDialog.setTitle("User Name - ${dataModel.firstName} ")
                alertDialog.setMessage("User email ${dataModel.email}")
                alertDialog.setCancelable(true).show()

                alertDialog.show()
            }
        }
    }
}